### ISSUE TITLE
Feature Request: Vault access via LDAP / Token

### ISSUE THREAD

**Issue by denysvitali** [hashicorp/waypoint#1531]
**Is your feature request related to a problem? Please describe.**

Currently the Vault plugin only allows the following two authentication methods:
- AWS
- Kubernetes

Whilst this might be great for some use-cases, it is not for some others.

**Describe the solution you'd like**

We are using Vault with Tokens and LDAP credentials, and we would thus like to be able to get the secrets from Vault through Waypoint with these additional two authentication methods:
- LDAP (Username + Password)
- Token

**Describe alternatives you've considered**

No alternatives are really possible. Either you use Vault with Kubernetes / AWS credentials, or you don't.

My current dirty workaround is to use a bash script to log-in into vault, do some magic with `jq` and return a `.env` file that I then load into our Waypoint plugin.

Obviously this is hackish, and needs to be replaced with a Waypoint implementation.

**Explain any additional use-cases**

Vault offers a lot of authentication methods, ideally in the future we should support them all.
Given the fact that Waypoint is an Hashicorp product, I would (as an user) expect to fully support the Hashicorp ecosystem, and thus, fully support Vault.

It's obvious that Waypoint is still not 1.0.0 yet and thus these features are missing, hence the creation of this issue to track the progress.

**Additional context**

N/A

---

Since I'm really interested into this thing I might work on a solution myself and make a PR soon, just fyi

**Comment by briancain** [comment]
Hey there @denysvitali - sounds great! Agreed, Waypoint could benefit from these improvements here.

Before you get started, a question I had was how you were envisioning the implementation here? Do you plan on updating the existing Vault plugin to support LDAP and tokens?

Otherwise sounds good, appreciate you taking the time to open an issue before starting the implementation ❤️

**Comment by denysvitali** [comment]
Definitely! I think that extending the current plugin is the best way to do it.


I'm open to alternatives though :)

**Comment by briancain** [comment]
Nope, that sounds good to me @denysvitali ! 👍🏻 Let us know if you need anything along the way
